source/GameData.cpp

```diff
@@ -797,43 +836,21 @@ void GameData::LoadSources()
 	for(const string &path : globalPlugins)
 	{
 		if(Files::Exists(path + "data") || Files::Exists(path + "images") || Files::Exists(path + "sounds"))
-			sources.push_back(path);
+			if(LoadAboutPlugin(path))
+				sources.push_back(path);
 	}
 
 	vector<string> localPlugins = Files::ListDirectories(Files::Config() + "plugins/");
 	for(const string &path : localPlugins)
 	{
 		if(Files::Exists(path + "data") || Files::Exists(path + "images") || Files::Exists(path + "sounds"))
-			sources.push_back(path);
+			if(LoadAboutPlugin(path))
+				sources.push_back(path);
 	}
 
-	// Load the plugin data, if any.
-	for(auto it = sources.begin() + 1; it != sources.end(); ++it)
-	{
-		// Get the name of the folder containing the plugin.
-		size_t pos = it->rfind('/', it->length() - 2) + 1;
-		string name = it->substr(pos, it->length() - 1 - pos);
-
-		// Load the about text and the icon, if any.
-		plugins[name] = Files::Read(*it + "about.txt");
-
-		// Create an image set for the plugin icon.
-		auto icon = make_shared<ImageSet>(name);
-
-		// Try adding all the possible icon variants.
-		if(Files::Exists(*it + "icon.png"))
-			icon->Add(*it + "icon.png");
-		else if(Files::Exists(*it + "icon.jpg"))
-			icon->Add(*it + "icon.jpg");
-
-		if(Files::Exists(*it + "icon@2x.png"))
-			icon->Add(*it + "icon@2x.png");
-		else if(Files::Exists(*it + "icon@2x.jpg"))
-			icon->Add(*it + "icon@2x.jpg");
-
-		icon->ValidateFrames();
-		spriteQueue.Add(icon);
-	}
+	// Freeze the current plugin state to check if plugin state has changed when
+	// toggling user settings.  This should only be called once per instance.
+	Plugins::Freeze();
```

Comment:
I don't think you need this anymore?

Reply:
This has been deleted since implementing tehhowch's struct suggestion.